src/App.tsx

```diff
@@ -1,73 +1,122 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import debounce from 'lodash.debounce';
+import { Person } from './types/Person';
+
+const delay = 300;
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const people: Person[] = peopleFromServer;
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(true);
+  const [chosenPerson, setChosenPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, delay), []);
```

Comment:
The `debounce` function is correctly used with `useCallback`, but ensure that the dependencies array is correctly managed. If `setAppliedQuery` or `delay` changes, the `debounce` function should be recreated. Consider adding `setAppliedQuery` and `delay` to the dependencies array if they are expected to change.